dagmc_stats/DagmcQuery.py

```diff
@@ -99,5 +138,62 @@ def calc_tris_per_vert(self, ignore_zero=True):
         if self._vert_data.empty:
             self._vert_data = self._vert_data.append(t_p_v_data)
         else:
-            self._vert_data.set_index('vert_eh').join(
-                self._vert_data.append(t_p_v_data).set_index('vert_eh'))
+            self._vert_data.set_index('vert_eh').join(self._vert_data.append(t_p_v_data).set_index('vert_eh'))
+
+    def calc_triangle_aspect_ratio(self):
+        """
+        Calculate triangle aspect ratio data (according to the equation:
+        (abc)/(8(s-a)(s-b)(s-c)), where s = .5(a+b+c).)
+
+        inputs
+        ------
+        none
+
+        outputs
+        -------
+        none
+        """
+        t_a_r_data = []
+        tris = self.get_tris()
+        for tri in tris:
+            side_lengths = list(self.get_tri_side_length(tri).values())
+            s = .5*(sum(side_lengths))
+            top = np.prod(side_lengths)
+            bottom = 8*np.prod(s-side_lengths)
+            t_a_r = top/bottom
+            row_data = {'tri_eh': tri, 'aspect_ratio': t_a_r}
+            t_a_r_data.append(row_data)
+        if self._tri_data.empty:
+            self._tri_data = self._tri_data.append(t_a_r_data)
+        else:
+            self._tri_data.set_index('tri_eh').join(self._tri_data.append(t_a_r_data).set_index('tri_eh'))
```

Comment:
We have this same pattern in each method.  Should it be a small method of its own?
```
def update_tri_data(self, new_data)
     if self._tri_data.empty:
         self._tri_data = self._tri_data.append(new_data)
     else:
         self._tri_data.set_index('tri_eh').join(self._tri_data.append(new_data).set_index('tri_eh'))
```

Reply:
On a related note...  I don't think any of your tests are designed to test this update capability.  They all start from empty `DagmcQuery` objects.  Perhaps we need some new tests that:
• call the same query multiple times on the same meshset => the database should not grow
• call the same query multiple times on different mesh sets => the database should grow appropriately 
• call different queries on the same meshsets
• call different queries on different meshsets

(Not sure what the expected behavior is for the last two)

Reply:
regarding these tests:

* call the same query multiple times on the same meshset => the database should not grow --> we need to add a check to make sure we are not adding a second column (currently it does)
* call the same query multiple times on different mesh sets => the database should grow appropriately --> you can't call multiple queries on different meshsets. Each set of meshsets gets its own query object so this action isn't a valid test. It's equivalent to calling a single query each on two different query objects.
* call different queries on the same meshsets --> this is what you probably meant in the last point. One query object with multiple different queries should grow (one test exists now to show that)
* call different queries on different meshsets --> this is the same as point 2 above